src/pytest_check/check_raises.py

```diff
@@ -97,3 +106,6 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             # without raising an error, hence `return True`.
             log_failure(self.msg if self.msg else exc_val)
             return True
+
+        # Stop on fail, so return True
+        return False
```

Comment:
This was a bit of a guess, but reading the comments, returning False seemed like the right thing to do...